opencl-aot/source/main.cpp

```diff
@@ -253,17 +265,46 @@ getCompilerBuildLog(const CLProgramUPtr &ProgramUPtr, cl_device_id Device) {
 
 int main(int Argc, char *Argv[]) {
   // step 0: set command line options
-  cl::opt<std::string> OptInputBinary(
-      cl::Positional, cl::Required,
-      cl::desc("<input SPIR-V or OpenCL program binary>"),
+  cl::opt<std::string> OptInputFile(
+      cl::Positional,
+      cl::desc("<input SPIR-V or OpenCL program binary or OpenCL C source>"),
```

Comment:
Not sure if it should be OpenCL program **binary** or OpenCL program **object**

Let's summarize it (please fix me if I was wrong somewhere):

| Input  | Command | Output |
| ------------- | ------------- | ------------- |
| OpenCL C source | build | OpenCL program binary |
| OpenCL C source | compile | OpenCL program object |
| OpenCL C source  | link | **Not valid, show error message** |
| SPIR-V | build | OpenCL program binary |
| SPIR-V | compile | **Not valid, show error message** |
| SPIR-V  | link | OpenCL program binary |
| OpenCL compiled program object | build | OpenCL program binary |
| OpenCL compiled program object | compile | **Not valid, show error message** |
| OpenCL compiled program object | link | OpenCL program binary |
| LLVM IR | build | **The same as for SPIR-V?** |
| LLVM IR | compile | **The same as for SPIR-V?** |
| LLVM IR  | link | **The same as for SPIR-V?** |
| OpenCL program binary | build | **Is it valid use case? Will we get the same program binary?** |
| OpenCL program binary | compile | **Not valid, show error message** |
| OpenCL program binary | link | **Is it valid use case? If it's not valid, show error message** |

Reply:
> SPIR-V | compile | Not valid, show error message
> -- | -- | --

This is a valid call of OpenCL API. You can "compile" a SPIR-V file and then use link API to merge several SPIR-V files together

Reply:
> OpenCL compiled program object | compile | Not valid, show error message
> -- | -- | --

This is also valid. The output of compiling a OpenCL object file is itself.

Reply:
Thanks @AlexeySachkov and @linkren. @linkren , can you please include missing workflows to the source code?

Reply:
They are already there. There is no specific error message but internal returned error is displayed for invalid workflows. Users should know the type of input file they feed to this tool.

Reply:
They are already there. There is no specific error message but internal returned error is displayed for invalid workflows. Users should know the type of input file they feed to this tool.